### PR TITLE
[relates #114] DB Connector e2e tests. part 1

### PR DESCRIPTION
**Pull request by Stefan365**
DB connector test, part 1 - START-FINISH connections.

### CODE REVIEW

File: ui-tests/src/test/java/io/syndesis/qe/pages/SyndesisPageObject.java [hunk 1]
```diff
@@ -168,4 +170,22 @@ public String getElementText(By locator) {
...
+				condition = visible;
+				break;
+			case "Disabled":
+			case "Inactive":
+				condition = disabled;
+			default:
+				condition = visible;
+				break;
+		}
+		SelenideElement buttonElement = this.getButton(buttonTitle).shouldBe(condition);
```
Comment (dsimansk) [inline ui-tests/src/test/java/io/syndesis/qe/pages/SyndesisPageObject.java]: Is there any benefit to have such a method. Imho you always can use getButton() and inject the condition you need in a appropriate code.

File: ui-tests/src/test/java/io/syndesis/qe/steps/integrations/IntegrationSteps.java [hunk 2]
```diff
@@ -288,6 +290,15 @@ public void fillActionConfigureField(String fieldId, String value) {
...
 
+//	@Then("^she fills query input with \"([^\"]*)\" value$")
+//	public void fillSQLquery(String query) {
+//		editPage.getPeriodicSqlComponent().fillSQLquery(query);
+//	}
+//
+//	@Then("^she fills period input with \"([^\"]*)\" value$")
+//	public void fillSQLperiod(String period) {
+//		editPage.getPeriodicSqlComponent().fillSQLperiod(period);
+//	}
```
Comment (dsimansk) [inline ui-tests/src/test/java/io/syndesis/qe/steps/integrations/IntegrationSteps.java]: This seems OK.

File: ui-tests/src/test/java/io/syndesis/qe/pages/integrations/edit/ActionConfigureComponentInvokeSql.java [hunk 3]
```diff
@@ -0,0 +1,32 @@
...
+
+import org.openqa.selenium.By;
+
+import lombok.extern.slf4j.Slf4j;
+
+/**
+ * Created by sveres on 12/7/17.
+ */
+@Slf4j
+public class ActionConfigureComponentInvokeSql extends ActionConfigureComponent {
```
Comment (dsimansk) [inline ui-tests/src/test/java/io/syndesis/qe/pages/integrations/edit/ActionConfigureComponentInvokeSql.java]: Maybe a parent class with defined `inputFill` method and then specify them to Invoke and Periodic flavour?

File: ui-tests/src/test/java/io/syndesis/qe/pages/SyndesisPage.java [hunk 4]
```diff
@@ -20,7 +21,9 @@
 	SELECT_START_CONNECTION(new IntegrationConnectionSelectComponentStart()),
 	SELECT_FINISH_CONNECTION(new IntegrationConnectionSelectComponentFinish()),
 	CHOOSE_A_FINISH_CONNECTION(new IntegrationConnectionSelectComponentFinish()),
-	ADD_TO_INTEGRATION(new IntegrationSaveOrAddStepComponent());
+	ADD_TO_INTEGRATION(new IntegrationSaveOrAddStepComponent()),
+	PERIODIC_SQL_INVOCATION(new ActionConfigureComponentPeriodicSql()),
+	INVOKE_SQL(new ActionConfigureComponentPeriodicSql()),;
```
Comment (lvydra) [inline ui-tests/src/test/java/io/syndesis/qe/pages/SyndesisPage.java]: There should be probably different class.

File: ui-tests/src/test/java/io/syndesis/qe/pages/SyndesisPageObject.java (hunk 1 above)
Comment (Stefan365) [reply ui-tests/src/test/java/io/syndesis/qe/pages/SyndesisPageObject.java]: no problem. method removed.

File: ui-tests/src/test/java/io/syndesis/qe/pages/integrations/edit/ActionConfigureComponentInvokeSql.java (hunk 3 above)
Comment (Stefan365) [reply ui-tests/src/test/java/io/syndesis/qe/pages/integrations/edit/ActionConfigureComponentInvokeSql.java]: you mean abstract inputFill method in parent?

File: ui-tests/src/test/java/io/syndesis/qe/pages/SyndesisPage.java (hunk 4 above)
Comment (Stefan365) [reply ui-tests/src/test/java/io/syndesis/qe/pages/SyndesisPage.java]: @lvydra could you be more specific, please.

Comment (dsimansk) [reply ui-tests/src/test/java/io/syndesis/qe/pages/SyndesisPage.java]: I guess this line `INVOKE_SQL(new ActionConfigureComponentPeriodicSql()),;`
should be `INVOKE_SQL(new ActionConfigureComponentInvokeSql()),;`

File: ui-tests/src/test/java/io/syndesis/qe/pages/integrations/edit/ActionConfigureComponentInvokeSql.java (hunk 3 above)
Comment (dsimansk) [reply ui-tests/src/test/java/io/syndesis/qe/pages/integrations/edit/ActionConfigureComponentInvokeSql.java]: Yes, to have parent with all the common code and then Periodic child with one additional method to specify interval.

File: ui-tests/src/test/java/io/syndesis/qe/pages/SyndesisPageObject.java (hunk 1 above)
Comment (Stefan365) [reply ui-tests/src/test/java/io/syndesis/qe/pages/SyndesisPageObject.java]: just to be sure what you mean. 
st. like "..shouldBe(Condition.attribute(status))"?
(where 'status' is String value from cucumber step)

Comment (dsimansk) [reply ui-tests/src/test/java/io/syndesis/qe/pages/SyndesisPageObject.java]: I haven't realized that you consume status from step also. In that case it's ok. But maybe I would refactor it to more generic approach, like `conditionValueOf(String status)` that return selenide Conditions based on provided string to some util class. To benefit from this conversion also elsewhere not just on button elements.

**Comment by Stefan365** [comment]
includes in PR #224

**Pull request closed without merging**